Fatras/include/ActsFatras/Physics/PhotonConversion/PhotonConversion.hpp

```diff
@@ -0,0 +1,311 @@
+#include "Acts/Definitions/Units.hpp"
+#include "Acts/Material/MaterialSlab.hpp"
+#include "Acts/Utilities/UnitVectors.hpp"
+#include "ActsFatras/EventData/Particle.hpp"
+#include "ActsFatras/EventData/ProcessType.hpp"
+#include "ActsFatras/Utilities/ParticleData.hpp"
+
+#include <limits>
+#include <random>
+#include <vector>
+
+#include <math.h>
+
+namespace ActsFatras {
+/// This class handles the photon conversion. It evaluates the distance
+/// after which the interaction will occur and the final state due the
+/// interaction itself.
+class PhotonConversion {
+ public:
+  using Scalar = ActsFatras::Particle::Scalar;
+  /// Lower energy threshold for produced children
+  Scalar minimumMomentum = 100. * Acts::UnitConstants::MeV;
+  /// Scaling factor of children energy
+  Scalar childEnergyScaleFactor = 2.;
+  /// Scaling factor for photon conversion probability
+  Scalar conversionProbScaleFactor = 0.98;
+
+  /// Method for evaluating the distance after which the photon
+  /// conversion will occur.
+  ///
+  /// @tparam generator_t Type of the random number generator
+  /// @param [in, out] generator The random number generator
+  /// @param [in] momentum The momentum of the particle
+  ///
+  /// @return valid X0 limit and no limit on L0
+  template <typename generator_t>
+  std::pair<Scalar, Scalar> generatePathLimits(generator_t& generator,
+                                               const Particle& particle) const;
+
+  /// This method evaluates the final state due to the photon conversion.
+  ///
+  /// @tparam generator_t Type of the random number generator
+  /// @param [in, out] generator The random number generator
+  /// @param [in, out] particle The interacting photon
+  /// @param [out] generated List of generated particles
+  ///
+  /// @return True if the conversion occured, else false
+  template <typename generator_t>
+  bool run(generator_t& generator, Particle& particle,
+           std::vector<Particle>& generated) const;
+
+ private:
+  /// This method constructs and returns the child particles.
+  ///
+  /// @param [in] photon The interacting photon
+  /// @param [in] childEnergy The energy of one child particle
+  /// @param [in] childDirection The direction of the child particle
+  ///
+  /// @return Array containing the produced leptons
+  std::array<Particle, 2> generateChildren(
+      const Particle& photon, Scalar childEnergy,
+      const Particle::Vector3& childDirection) const;
+
+  /// Generate the energy fraction of the first child particle.
+  ///
+  /// @tparam generator_t Type of the random number generator
+  /// @param [in, out] generator The random number generator
+  /// @param [in] gammaMom The momentum of the photon
+  ///
+  /// @return The energy of the child particle
+  template <typename generator_t>
+  Scalar generateFirstChildEnergyFraction(generator_t& generator,
+                                          Scalar gammaMom) const;
+
+  /// Generate the direction of the first child particle.
```

Comment:
```suggestion
  /// Generate the direction of the child particles.
```
The simulation uses the approximation of zero recoil momentum such that the two electron share the same direction, correct?

Reply:
The first direction is calculated within this function but the second one is calculated from the electron and photon momentum. So depending on what you mean with share, both leptons don't have to fly in the same direction.

Reply:
But the two particles you create use the same correction, as stated [here](https://github.com/acts-project/acts/pull/597/files#diff-5c826f86f341a3323c50953efc3756bcca961e527a3215b2c69b45437c02ac30R274). I read that such that they are forced to fly in the same direction.

Reply:
Sorry, I didn't check it again and had an older version in mind. Indeed, they are the same direction. I will fix it.

Reply:
Done